hack/scripts/validate_state.sh

```diff
@@ -0,0 +1,105 @@
+#!/bin/bash
+function find_in_array() {
+    for i in $1
+    do
+        if [ "$i" == "$2" ] ; then
+            return 0
+        fi
+    done
+    return 1
+}
+
+for node in $(kubectl get nodes -o name);
+do
+    echo "Current : $node"
+    node_name="${node##*/}"
+    node_ip=$(kubectl get "$node"  -o jsonpath='{$.status.addresses[?(@.type=="InternalIP")].address}')
+    echo "Node internal ip: $node_ip"
+    privileged_pod=$(kubectl get pods -n kube-system -l app=privileged-daemonset -o wide | grep "$node_name" | awk '{print $1}')
+    echo "privileged pod : $privileged_pod"
+    while ! [ -f "azure_endpoints.json" ]
+    do
+        echo "trying to get the azure_endpoints"
+        kubectl exec -i "$privileged_pod" -n kube-system -- bash -c "cat /var/run/azure-cns/azure-endpoints.json" > azure_endpoints.json
+        if [ -f "azure_endpoints.json" ] && [ ! -s "azure_endpoints.json" ]; then
+            rm azure_endpoints.json
+        fi
+        sleep 10
+    done
+
+    cilium_agent=$(kubectl get pod -l k8s-app=cilium -n kube-system -o wide | grep "$node_name" | awk '{print $1}')
+    echo "cilium agent : $cilium_agent"
+    
+    while ! [ -f "cilium_endpoints.json" ]
+    do
+        echo "trying to get the cilium_endpoints"
+        kubectl exec -i "$cilium_agent" -n kube-system -- bash -c "cilium endpoint list -o json" > cilium_endpoints.json
+        if [ -f "cilium_endpoints.json" ] && [ ! -s "cilium_endpoints.json" ]; then
+            rm cilium_endpoints.json
+        fi
```

Comment:
I think you could simplify this by moving the "is file empty" check into the `while` condition:
```
while [ ! -f "cilium_endpoints.json" ] || [ ! -s "cilium_endpoints.json" ]; do
    ...
    # no need to `rm` cilium_endpoints.json, kubectl will overwrite it on retry anyway
done
```